ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/ComparisonInterpreter.java

```diff
@@ -118,6 +120,12 @@ private static ComparisonNullCheckFunction getNullCheckFunction(
       return Optional.of((o1, o2) -> castLeft.cast(o1).compareTo(castRight.cast(o2)));
     } else if (leftType == SqlBaseType.STRING) {
       return Optional.of((o1, o2) -> o1.toString().compareTo(o2.toString()));
+    } else if (either(leftType, rightType, SqlBaseType.BYTES)) {
```

Comment:
Does this mean you can compare BYTES with other types?

Reply:
It will eventually throw an error if either of them aren't BYTES. Updated to check that condition earlier.